ui/app/src/components/SiteToolsApp/SiteToolsApp.tsx

```diff
@@ -157,7 +163,18 @@ export function SiteToolsApp(props: SiteToolsAppProps) {
         {activeToolId ? (
           tool ? (
             <Suspencified>
-              <Widget {...tool} extraProps={{ embedded: false, showAppsButton, onSubmittingAndOrPendingChange }} />
+              <Widget
+                {...tool}
+                extraProps={{
+                  embedded: false,
+                  mountMode,
+                  showAppsButton,
+                  onSubmittingAndOrPendingChange,
+                  onMinimize: () => {
+                    dispatch(updateWidgetDialog({ isMinimized: true }));
+                  }
```

Comment:
Could we send this in from the parent component so we don't couple SiteToolsApp to redux?